igvc/src/filter_lidar/main.cpp

```diff
@@ -0,0 +1,43 @@
+#include <pcl/point_types.h>
+#include <pcl_ros/point_cloud.h>
+#include <ros/publisher.h>
+#include <ros/ros.h>
+
+ros::Publisher _pointcloud_pub;
+
+double x_offset, y_offset, y_size, x_size;
+
+void point_cloud_callback(const pcl::PointCloud<pcl::PointXYZ>::ConstPtr& msg)
+{
+  pcl::PointCloud<pcl::PointXYZ> result;
+  for (auto it = msg->points.begin(); it != msg->points.end(); it++)
+  {
+    if (!((it->x + x_offset < x_size && it->x + x_offset > -x_size) &&
```

Comment:
IMO this should be a function for figuring out if something is inside a rect (do you have a geometry library?), but you coul also do

`abs(it->x + x_offset < x_size)` as well to make this a bit more readable.